app/services/daily_digest_scheduler_service.rb

```diff
@@ -0,0 +1,47 @@
+class DailyDigestSchedulerService
+  attr_reader :digest_range_hour
+  attr_accessor :starts_at, :ends_at
+
+  LOCK_NAME = "daily_digest_scheduler".freeze
+
+  def self.call(*args)
+    new.call(*args)
+  end
+
+  def initialize
+    @digest_range_hour = ENV.fetch("DIGEST_RANGE_HOUR", 8).to_i
+  end
+
+  def call(starts_at: nil, ends_at: nil)
+    starts_at = starts_at || configured_starts_at
+    ends_at = ends_at || configured_ends_at
+
+    digest_range = nil
+
+    run_with_advisory_lock do
+      digest_range = DigestRange.find_or_initialize_by(
+        starts_at: starts_at, ends_at: ends_at
+      )
+      return if digest_range.persisted?
+    end
+
+    digest_range.save!
```

Comment:
Is there any risk of a clash here? The `save` runs outside of the lock, which means there's a point where we've released the lock, but no record is persisted and another box could, in theory execute `find_or_initialize_by` before we `save`? Or is that not how the advisory lock stuff works?

I also vaguely wondered how we feel about a unique constraint at the database level which would add an extra level of security against duplicate rows?